packages/replay/metadata/index.js

```diff
@@ -0,0 +1,42 @@
+const { maybeLog } = require("../src/utils");
+const test = require("./test");
+
+// Each known metadata block should have a sanitizer that will check the contents before the upload
+const handlers = {
+  test: test.validate
+};
+
+const ALLOWED_KEYS = Object.keys(handlers);
+
+// Sanitizing arbitrary recording metadata before uploading by removing any
+// non-object values (allowing null) and limiting object values to known keys or
+// userspace keys prefixed by `x-`.
+function sanitize(metadata, opts = {}) {
+  const updated = {};
+  Object.keys(metadata).forEach((key) => {
+    const value = metadata[key];
+    // intentionally allowing `null` with the `typeof` check here
+    if (typeof value === "object") {
+      if (value === null || key.startsWith("x-")) {
```

Comment:
I'm not sure how I feel about the `x-` scheme. I have two questions: 
1) what are the trade-offs vs. allowing a particular key like `extras` and allowing whatever to be nested under that?
2) So people will be able to nest as much as they want under any key that starts with `x-`? What kind of use cases motivate us to have this right now? I'm all for making sure that we have custom metadata eventually, I just want to make sure we don't paint ourselves into a corner by choosing a particular schema before we know how people will actually use it and then having to support that for the rest of time.

Reply:
1. The idea here was to reserve space for us to grow custom metadata in the future without stepping on userspace fields. Since this isn't public API yet, we don't expect to see much if any of this but it seems useful to carve out a space for custom metadata now.
2. That's fair. I'm not sure how people will use it but I'm also not sure how we'll figure it out if we don't provide a place. Again, this isn't public yet so we can walk it back a bit if we find reasons to do so.